File: C1.UWP.OrgChart/VB/OrgChartSamples/PersonTemplateSelector.vb
Imports Windows.UI.Xaml
Imports System.Threading.Tasks
Imports System.Text
Imports System.Linq
Imports System.Collections.Generic
Imports System

''' <summary>
''' Class used to select the templates for items being created.
''' </summary>
Public Class PersonTemplateSelector
    Inherits Windows.UI.Xaml.Controls.DataTemplateSelector
    Protected Overrides Function SelectTemplateCore(item As Object, container As DependencyObject) As DataTemplate
        Dim p As Person = TryCast(item, Person)
        'var e = Application.Current.RootVisual as FrameworkElement;
        'return p.Position.IndexOf("Director") > -1
        '    ? e.Resources["_tplDirector"] as DataTemplate
        '    : e.Resources["_tplOther"] as DataTemplate;


        If p.Position.IndexOf(Strings.Director) > -1 Then
            Return DirectorTemplate
        ElseIf p.Position.IndexOf(Strings.Manager) > -1 Then
            Return ManagerTemplate
        ElseIf p.Position.IndexOf(Strings.Designer) > -1 Then
            Return DesignerTemplate
        Else
            Return OtherTemplate
        End If
    End Function

    Public Property DirectorTemplate() As DataTemplate
    Public Property ManagerTemplate() As DataTemplate
    Public Property DesignerTemplate() As DataTemplate
    Public Property OtherTemplate() As DataTemplate
End Class

File: C1.UWP.OrgChart/CS/OrgChartSamples/PersonTemplateSelector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Controls;

namespace OrgChartSamples
{
    /// <summary>
    /// Class used to select the templates for items being created.
    /// </summary>
    public class PersonTemplateSelector : DataTemplateSelector
    {
        protected override DataTemplate SelectTemplateCore(object item, DependencyObject container)
        {
            var p = item as Person;
            //var e = Application.Current.RootVisual as FrameworkElement;
            //return p.Position.IndexOf("Director") > -1
            //    ? e.Resources["_tplDirector"] as DataTemplate
            //    : e.Resources["_tplOther"] as DataTemplate;


            if (p.Position.IndexOf(Strings.Director) > -1)
            {
                return DirectorTemplate;
            }
            else if (p.Position.IndexOf(Strings.Manager) > -1)
            {
                return ManagerTemplate;
            }
            else if (p.Position.IndexOf(Strings.Designer) > -1)
            {
                return DesignerTemplate;
            }
            else
            {
                return OtherTemplate;
            }
        }

        public DataTemplate DirectorTemplate { get; set; }
        public DataTemplate ManagerTemplate { get; set; }
        public DataTemplate DesignerTemplate { get; set; }
        public DataTemplate OtherTemplate { get; set; }
    }
}
